kratos/python_scripts/analysis_stage.py

```diff
@@ -124,12 +124,22 @@ def InitializeSolutionStep(self):
         """This function performs all the required operations that should be executed
         (for each step) BEFORE solving the solution step.
         """
+        # We reinitialize if remeshed previously
+        if self._CheckIfModelIsModified():
+            self.ClearDatabase()
+            self.ReInitializeSolver()
```

Comment:
I am quite convinced this cannot be in the baseclass

Reply:
Well, I am quite sure we agreed this in order to be used with the corresponding NR for an adaptive strategy